src/olm/session/mod.rs

```diff
@@ -207,17 +212,25 @@ impl Session {
         if self.has_received_message() {
             OlmMessage::Normal(message)
         } else {
-            let message = PreKeyMessage::new(
-                self.session_keys.one_time_key,
-                self.session_keys.base_key,
-                self.session_keys.identity_key,
-                message,
-            );
+            let message = PreKeyMessage::new(self.session_keys, message);
 
             OlmMessage::PreKey(message)
         }
     }
 
+    /// Get the keys associated with this session.
+    pub fn session_keys(&self) -> SessionKeys {
+        self.session_keys
+    }
+
+    /// Get the [`MessageKey`] to encrypt the next message.
+    ///
+    /// **Note**: This *must* be used to encrypt the message.
```

Comment:
Perhaps something like:

> **Note**: This *should* be used to encrypt the next message, `Session`s support only a limited amount of out of order messages before decryption starts to fail.

Reply:
I *think* the intent is to express this, so how about something a bit more detailed?

```suggestion
    /// **Note**: Each key obtained in this way should be used to encrypt
    /// a message and the message must then be sent to the recipient.
    ///
    /// Failing to do so will increase the number of out-of-order messages on
    /// the recipient side. Given that a `Session` can only support a limited
    /// number of out-of-order messages, this will eventually lead to
    /// undecryptable messages.
```

Reply:
Love it, thank you

Reply:
Hilariously github has attributed that suggestion to @poljar.... https://github.com/matrix-org/vodozemac/pull/49/commits/0f1d1e530f6f199bb4ba884aa0be0c5bc21aa429

Reply:
As it should, thanks github.